src/hierarchies/fa_hierarchy.cc

```diff
@@ -67,9 +68,16 @@ void FAHierarchy::initialize_state() {
   state.eta = Eigen::MatrixXd::Zero(card, hypers->q);
   state.lambda = Eigen::MatrixXd::Zero(dim, hypers->q);
   state.psi_inverse = state.psi.cwiseInverse().asDiagonal();
-  state.cov_chol = (state.lambda * state.lambda.transpose() +
-                    Eigen::MatrixXd(state.psi.asDiagonal()))
-                       .llt();
+  Eigen::MatrixXd temp_chol =
+      (state.lambda.transpose() * state.psi_inverse * state.lambda +
+       Eigen::MatrixXd::Identity(hypers->q, hypers->q))
+          .llt()
+          .matrixL()
+          .solve(Eigen::MatrixXd::Identity(hypers->q, hypers->q));
+  state.cov_logdet =
+      -2 * Eigen::MatrixXd(temp_chol).diagonal().array().log().sum() +
+      state.psi.array().log().sum();
+  state.cov_wood = temp_chol * state.lambda.transpose() * state.psi_inverse;
```

Comment:
I have seen this piece of code multiple times already in these files. It would be a good idea to create one or more methods to encapsulate these lines, e.g. which compute the log-determinant and/or the `cov_wood` factor

Reply:
Yes it would be better for sure, I'm implementing it right now